src/System.Text.Json/tests/Serialization/ReadValueTests.cs

```diff
@@ -493,5 +494,76 @@ public static void ReadPartial()
             obj = JsonSerializer.Deserialize(ref reader, typeof(int));
             Assert.Equal(1, obj);
         }
+
+        [Theory]
+        [InlineData("0,1")]
+        [InlineData("0 1")]
+        [InlineData("0 {}")]
+        public static void TooMuchJson(string json)
+        {
+            byte[] jsonBytes = Encoding.UTF8.GetBytes(json);
+
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<int>(json));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<int>(jsonBytes));
+            Assert.Throws<JsonException>(() => JsonSerializer.DeserializeAsync<int>(new MemoryStream(jsonBytes)).Result);
+
+            // Using a reader directly doesn't throw.
+            Utf8JsonReader reader = new Utf8JsonReader(jsonBytes);
+            JsonSerializer.Deserialize<int>(ref reader);
+            Assert.NotEqual(jsonBytes.Length, reader.BytesConsumed);
+        }
+
+        [Theory]
+        [InlineData("0/**/")]
+        [InlineData("0 /**/")]
+        public static void TooMuchJsonWithComments(string json)
+        {
+            byte[] jsonBytes = Encoding.UTF8.GetBytes(json);
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<int>(json));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<int>(jsonBytes));
+            Assert.Throws<JsonException>(() => JsonSerializer.DeserializeAsync<int>(new MemoryStream(jsonBytes)).Result);
```

Comment:
Something is off here.  `Task<T>.Result` will always wrap exceptions in an AggregateException.  So either this test should fail, because it'll be throwing an AggregateException rather than a JsonException, or the exception is actually emerging synchronously from the DeserializeAsync call, which it really shouldn't do.

Reply:
It appears the async method being called returns `ValueTask<T>` (instead of `Task<T>`) which has this exception behavior of not returning `AggregateException`. Bug?

Reply:
Ah, if it's returning ValueTask, then this makes sense. Thanks.